include/uk/event.h

```diff
@@ -267,8 +267,8 @@ static inline int uk_raise_event_ptr(struct uk_event *e, void *data)
 
 	uk_event_handler_foreach(itr, e) {
 		__uk_event_assert(*itr);
-		ret |= ((*itr)(data));
-		if (ret == UK_EVENT_HANDLED)
+		ret = ((*itr)(data));
+		if (ret < 0 || ret == UK_EVENT_HANDLED)
```

Comment:
Hey @mschlumpp,

just wanted to test this while debugging another issue and noticed it does not work. A successful return must be sticky, which is not the case in your solution. E.g., you could have `HANDLED_CONT`->`NOT_HANDLED` and it would return `NOT_HANDLED` but should return `HANDLED(_CONT)`.

The code should look like this:
```C
		...
		rc = ((*itr)(data));
		if (rc < 0 || rc == UK_EVENT_HANDLED)
			return rc;
		ret |= rc;
	}
```

Reply:
Good point. However, were the enum values intended to be able to represent a bit-mask? In your proposed change the `raise_event` could also return `UK_EVENT_HANDLED_CONT | UK_EVENT_HANDLED == 3`.